### DISCUSSION TITLE
Query string manager

### DISCUSSION THREAD

**Discussion by amiranagram** [livewire/livewire#4499] [3. Feature ideas]
Hi, I wanna run this idea before I try and do anything.

I'd like to develop a query string manager which will most likely be just a separate class that will be instantiated inside components and will deal with all things query strings.

The problems I'm trying to solve with this:

* Mounting code complexity which each new feature
* Readability
* Implement caching of `getQueryString` function result. It gets called a few times and it looks like an expensive operation.

I cannot go more into details since I'm not sure yet what approach I want to take. A draft PR could be the first thing I do and then we can discuss more specific things.

@joshhanley What do you think? Sorry for pinging you, I wanna hear your opinion.

**Comment by joshhanley** [comment]
@amiranagram yeah agreed, the query string stuff needs some work.

We were actually talking about moving most of it into the front end, when that gets re-written to use Alpine, that way we're not trying to compile the uri/ querystring on the backend. But I'm not sure how much as there will still need to be a backend component for collecting the data and sending it to and from the front end.

I'll check with Caleb though and see what his thoughts are on it.

**Reply by calebporzio** [reply]
Yeah, we intend to move most of the query string tracking stuff to the front-end like Josh said. The goal is to avoid reconstructing the URL on the backend which is prone to lots of edge cases.

I'd say let's hold off on this rewrite for now, thanks for being willing @amiranagram!

**Reply by amiranagram** [reply]
Np @calebporzio, feel free to ping me if you need help with this. 🙂
